File: backend/shared/models/screenplay.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, List
from uuid import UUID
from enum import Enum

class SceneStatus(str, Enum):
    PENDING = "pending"
    GENERATING = "generating"
    COMPLETED = "completed"
    FAILED = "failed"

class SceneBase(BaseModel):
    scene_id: int
    narration: str
    image_prompt: str
    video_prompt: str
    character_description: Optional[str] = None

class SceneResponse(SceneBase):
    id: UUID
    screenplay_id: UUID
    image_url: Optional[str] = None
    video_url: Optional[str] = None
    status: SceneStatus
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class CharacterSheetBase(BaseModel):
    name: str
    description: Optional[str] = None

class CharacterSheetResponse(CharacterSheetBase):
    id: UUID
    screenplay_id: UUID
    combined_view_url: Optional[str] = None
    front_view_url: Optional[str] = None
    side_view_url: Optional[str] = None
    back_view_url: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True

class ScreenplayStatus(str, Enum):
    DRAFT = "draft"
    CONFIRMED = "confirmed"
    GENERATING = "generating"
    COMPLETED = "completed"
    FAILED = "failed"

class ScreenplayBase(BaseModel):
    title: str

class ScreenplayDraft(BaseModel):
    """生成剧本草稿请求"""
    task_id: UUID = Field(..., alias="taskId")
    prompt: str
    user_images: Optional[List[str]] = Field(None, alias="userImages")
    scene_count: int = Field(7, alias="sceneCount")
    character_count: int = Field(2, alias="characterCount")
    
    class Config:
        populate_by_name = True

class ScreenplayConfirm(BaseModel):
    """确认剧本请求"""
    feedback: Optional[str] = None

class ScreenplayUpdate(BaseModel):
    """更新剧本请求"""
    title: Optional[str] = None

class ScreenplayResponse(ScreenplayBase):
    id: UUID
    task_id: UUID
    user_id: UUID
    status: ScreenplayStatus
    scenes: List[SceneResponse] = []
    character_sheets: List[CharacterSheetResponse] = []
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: lib/features/screenplay/models/screenplay.dart
import 'scene.dart';
import 'character_sheet.dart';

enum ScreenplayStatus {
  draft,
  confirmed,
  generating,
  completed,
  failed;

  static ScreenplayStatus fromString(String status) {
    switch (status.toLowerCase()) {
      case 'confirmed':
        return ScreenplayStatus.confirmed;
      case 'generating':
        return ScreenplayStatus.generating;
      case 'completed':
        return ScreenplayStatus.completed;
      case 'failed':
        return ScreenplayStatus.failed;
      default:
        return ScreenplayStatus.draft;
    }
  }
}

class Screenplay {
  final String id;
  final String taskId;
  final String userId;
  final String title;
  final ScreenplayStatus status;
  final List<Scene> scenes;
  final List<CharacterSheet> characterSheets;
  final DateTime createdAt;
  final DateTime updatedAt;
  
  Screenplay({
    required this.id,
    required this.taskId,
    required this.userId,
    required this.title,
    required this.status,
    this.scenes = const [],
    this.characterSheets = const [],
    required this.createdAt,
    required this.updatedAt,
  });
  
  factory Screenplay.fromJson(Map<String, dynamic> json) {
    return Screenplay(
      id: json['id']?.toString() ?? '',
      taskId: json['taskId']?.toString() ?? json['task_id']?.toString() ?? '',
      userId: json['userId']?.toString() ?? json['user_id']?.toString() ?? '',
      title: json['title'] ?? '',
      status: ScreenplayStatus.fromString(json['status'] ?? 'draft'),
      scenes: json['scenes'] != null
          ? (json['scenes'] as List)
              .map((s) => Scene.fromJson(s))
              .toList()
          : [],
      characterSheets: json['characterSheets'] != null
          ? (json['characterSheets'] as List)
              .map((c) => CharacterSheet.fromJson(c))
              .toList()
          : (json['character_sheets'] != null
              ? (json['character_sheets'] as List)
                  .map((c) => CharacterSheet.fromJson(c))
                  .toList()
              : []),
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : (json['created_at'] != null
              ? DateTime.parse(json['created_at'])
              : DateTime.now()),
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'])
          : (json['updated_at'] != null
              ? DateTime.parse(json['updated_at'])
              : DateTime.now()),
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'taskId': taskId,
      'userId': userId,
      'title': title,
      'status': status.name,
      'scenes': scenes.map((s) => s.toJson()).toList(),
      'characterSheets': characterSheets.map((c) => c.toJson()).toList(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
  
  bool get isDraft => status == ScreenplayStatus.draft;
  bool get isConfirmed => status == ScreenplayStatus.confirmed;
  bool get isGenerating => status == ScreenplayStatus.generating;
  bool get isCompleted => status == ScreenplayStatus.completed;
  bool get isFailed => status == ScreenplayStatus.failed;
}
